test/Test.h

```diff
@@ -18,10 +18,14 @@
 #define TEST_MAIN(TestClass) \
 int main(int argc, char *argv[]) \
 { \
+  std::vector<const char*> new_argv(argv, argv + argc); \
```

Comment:
This looks quite hacky. Isn't there a way to pass in this extra flag "from the outside"?

Reply:
More importantly, what's the purpose of the change, both always using "Default" theme and the change to not pass the arguments to `QTest::qExec`?

Reply:
> More importantly, what's the purpose of the change, both always using "Default" theme and the change to not pass the arguments to `QTest::qExec`?
If a theme isn't passed in via command line arguments, the theme selector dialog is shown. Most unit tests don't expect this dialog to be present and will hang indefinitely waiting for the main dialog to open. The tests do not read the arguments passed in via QTest::qExec, so I did not propagate '--theme Default'  to QTest.

Reply:
How do you normally execute these tests? One by one, or the entire test suite at once? What's the exact command you run, and in which directory?

Reply:
I use the same steps as the CI workflow (https://github.com/gitahead/gitahead/blob/master/.github/workflows/build-stage.yml#L106)
```
cd build/release
ninja check
```

Reply:
FTR, this is the error that this patch fixes:

```
Start  1: bare_repo
1/13 Test  #1: bare_repo ........................   Passed    3.33 sec
Start  2: init_repo
2/13 Test  #2: init_repo ........................***Failed    6.06 sec
********* Start testing of TestInitRepo *********
Config: Using QtTest library 5.12.8, Qt 5.12.8 (x86_64-little_endian-lp64 shared (dynamic) release build; by GCC 9.3.0)
FAIL!  : TestInitRepo::initTestCase() 'mWindow && qWaitForWindowActive(mWindow)' returned FALSE. ()
Loc: [../../test/init_repo.cpp(82)]
PASS   : TestInitRepo::cleanupTestCase()
Totals: 1 passed, 1 failed, 0 skipped, 0 blacklisted, 5945ms
********* Finished testing of TestInitRepo *********
```

Reply:
I couldn't find a way to pass in this argument from the test framework (e.g. from `CMakeLists.txt` or so).

Another way to solve this would be to just write `"Default"` into the `Settings`, which is then read [here](https://github.com/gitahead/gitahead/blob/29fe2b739fe031e04d7cd2976cc069f8622cfc38/src/app/Theme.cpp#L647). Not sure if that's possible here in `Test.h`. It would look cleaner than tampering with the argv.